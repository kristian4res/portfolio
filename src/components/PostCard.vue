<script setup lang="ts">
import { defineProps } from 'vue'
import { Card } from 'primevue'
import { Button } from 'primevue'

defineProps({
  postTitle: {
    type: String,
    required: true,
  },
  buttonLink: {
    type: String,
    required: true,
  },
  coverImage: {
    type: String,
    default:
      'https://images.unsplash.com/photo-1453365607868-7deed8cc7d26?q=80&w=3570&auto=format&fit=crop&ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D',
  },
  coverImageAlt: {
    type: String,
    default: 'A Japanese shiba inu dog',
  },
})
</script>

<template>
  <Card class="w-[30rem] overflow-hidden flex-shrink-0 flex flex-col justify-between">
    <template #header>
      <img class="w-full max-h-[18rem] object-cover" :alt="coverImageAlt" :src="coverImage" />
    </template>
    <template #title>{{ postTitle }}</template>
    <template #content>
      <p class="m-0 text-wrap">
        <slot></slot>
      </p>
    </template>
    <template #footer>
      <div class="flex gap-4 mt-1">
        <Button
          as="a"
          label="Find out more"
          class="w-full"
          :href="buttonLink"
          target="_blank"
          rel="noopener"
        />
      </div>
    </template>
  </Card>
</template>
