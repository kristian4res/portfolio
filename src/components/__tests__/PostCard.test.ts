import { mount } from '@vue/test-utils'
import { describe, it, expect } from 'vitest'
import PostCard from '../PostCard.vue'

describe('PostCard.vue', () => {
  it('matches snapshot', () => {
    const wrapper = mount(PostCard, {
      props: {
        postTitle: 'Hello, world! This is a post card',
        buttonLink: 'https://google.com',
        buttonText: 'Google it!',
      },
      slots: {
        default: 'This is a test slot content',
      },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
