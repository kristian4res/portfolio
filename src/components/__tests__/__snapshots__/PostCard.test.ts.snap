// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PostCard.vue > matches snapshot 1`] = `
"<div class="p-card p-component w-[30rem] overflow-hidden flex-shrink-0 flex flex-col justify-between" buttontext="Google it!" data-pc-name="card" pc1="" data-pc-section="root">
  <div class="p-card-header" data-pc-section="header"><img class="w-full max-h-[18rem] object-cover" alt="A Japanese shiba inu dog" src="https://images.unsplash.com/photo-1453365607868-7deed8cc7d26?q=80&amp;w=3570&amp;auto=format&amp;fit=crop&amp;ixlib=rb-4.0.3&amp;ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D"></div>
  <div class="p-card-body" data-pc-section="body">
    <div class="p-card-caption" data-pc-section="caption">
      <div class="p-card-title" data-pc-section="title">Hello, world! This is a post card</div>
      <!---->
    </div>
    <div class="p-card-content" data-pc-section="content">
      <p class="m-0 text-wrap">This is a test slot content</p>
    </div>
    <div class="p-card-footer" data-pc-section="footer">
      <div class="flex gap-4 mt-1"><a class="p-button p-component w-full" aria-label="Find out more" data-pc-name="button" data-p-disabled="false" href="https://google.com" target="_blank" rel="noopener" pc2="" data-pc-section="root">
          <!----><span class="p-button-label" data-pc-section="label">Find out more</span>
          <!---->
        </a></div>
    </div>
  </div>
</div>"
`;
